app/charges/models.py

```diff
@@ -12,6 +12,36 @@
 ##
 ## @brief      Charges Model.
 ##
+
+from sqlalchemy.ext.mutable import Mutable
+
+## 
+## Class to create a mutable list in postgreSQL.
+## This is used for the progress_notes column
+##
+class MutableList(Mutable, list):
+    def append(self, value):
+        list.append(self, value)
+        self.changed()
+
+    def edit(self, index, value):
+        list.pop(self, index)
+        list.insert(self, index, value)
+        self.changed()
+    
+    def pop(self, index):
+        list.pop(self, index)
```

Comment:
This index is coming from the id of the note object, so we have to make sure the id is not greater than the length of the notes list. Do you have an idea for how the frontend will know what id to pass for a new note?

Reply:
So the way i set it up a new note will always have the largest id in the array, and will be appended to the end. When the client selects a note to edit or delete it will pass the note_id from the frontend for the findIndex() method.